pkg/kwokctl/runtime/kind/cluster_snapshot.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kind
+
+import (
+	"context"
+	"os"
+
+	"sigs.k8s.io/kwok/pkg/kwokctl/utils"
+	"sigs.k8s.io/kwok/pkg/kwokctl/vars"
+)
+
+// SnapshotSave save the snapshot of cluster
+func (c *Cluster) SnapshotSave(ctx context.Context, path string) error {
+	etcdContainerName, err := c.getComponentName("etcd")
+	if err != nil {
+		return err
+	}
+	kindName, err := c.getClusterName()
+	if err != nil {
+		return err
+	}
+
+	// Save to /var/lib/etcd/snapshot.db on container of Kind use Kubectl
+	// Why this path, etcd is just the volume /var/lib/etcd/ in container of Kind.
+	tmpFile := "/var/lib/etcd/snapshot.db"
+	err = c.KubectlInCluster(ctx, utils.IOStreams{}, "exec", "-i", "-n", "kube-system", etcdContainerName, "--", "etcdctl", "snapshot", "save", tmpFile, "--endpoints=127.0.0.1:2379", "--cert=/etc/kubernetes/pki/etcd/server.crt", "--key=/etc/kubernetes/pki/etcd/server.key", "--cacert=/etc/kubernetes/pki/etcd/ca.crt")
```

Comment:
nit: refactor `/etc/kubernetes/pki/etcd/` to a constant so that its not duplicated.

Reply:
also, why do we use `etcdctl` here while for `SnapshotRestore` we do `etcdctlPath := utils.PathJoin(bin, "etcdctl"+vars.BinSuffix)`?

Reply:
The save operation requires
1. Connect to the port to get the data and save it as backup file

The restore operation requires
1. restore the backup file to the db directory
2. stop etcd
3. replace the db directory
4. start etcd

The etcd image does not have `mv`, `cp` and `tar` commands, `tar` is required by `kubectl cp`.
So I use the host's etcdctl to extract the backup file to the db directory and then use `docker cp` to move it to the desired location